src/modules/user_grants.json

```diff
@@ -0,0 +1,534 @@
+                "type": "object",
+                "properties": {
+                    "app": {
+                        "$ref": "#/components/schemas/AppInfo"
+                    },
+                    "state": {
+                        "$ref": "#/components/schemas/GrantState"
+                    },
+                    "capability": {
+                        "$ref": "https://meta.comcast.com/firebolt/capabilities#/definitions/Capability"
+                    },
+                    "role": {
+                        "$ref": "https://meta.comcast.com/firebolt/capabilities#/definitions/Role"
+                    },
+                    "lifespan": {
+                        "type": "string",
+                        "enum": [
+                            "once",
+                            "forever",
+                            "appActive",
+                            "powerActive",
+                            "seconds"
+                        ]
+                    },
+                    "expires": {
+                        "type": "string",
+                        "format": "date-time"
+                    }
+                },
+                "additionalProperties": false,
+                "required": [
+                    "state",
+                    "capability",
+                    "role",
+                    "lifespan"
+                ],
+                "examples": [
+                    {
+                        "app": {
+                            "id": "certapp",
+                            "title": "Firebolt Certification"
+                        },
+                        "state": "active",
+                        "capability": "xrn:firebolt:capability:data:app-usage",
+                        "role": "use",
+                        "lifespan": "seconds",
+                        "expires": "2022-12-14T20:20:39+00:00"
+                    }
+                ]
+            },
+            "AppInfo": {
+                "description": "Information about an app that a grant was for",
+                "type": "object",
+                "properties": {
+                    "id": {
+                        "type": "string"
+                    },
+                    "title": {
+                        "type": "string"
+                    }
+                },
+                "additionalProperties": false,
+                "required": [
+                    "id"
+                ]
+            },
+            "GrantState": {
+                "description": "The state the grant is in",
+                "type": "string",
+                "enum": [
+                    "active",
+                    "denied"
+                ]
+            },
+            "GrantModificationOptions": {
+                "description": "Options when modifying any grant",
+                "type": "object",
+                "properties": {
+                    "appId": {
```

Comment:
Seem to have only appId are we going to have more props in future feels appId seems more appropriate given grant/deny and request definitely need appId

Reply:
> appId seems more appropriate given grant/deny and request definitely need appId

grant and deny do not need appId if it is a device-scoped grant.

Using an options object so it is extensible in the future, but currently only appId is optional.